volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,690 @@
+    """
+    An enumeration that creates a helpful mapping of opaque Windows 8 Amcache
+    'Program' subkey value names to their human-readable equivalent.
+    """
+
+    Product = "0"
+    Version = "1"
+    Publisher = "2"
+    InstallTime = "a"
+    MSIProductCode = "11"
+    MSIPackageCode = "12"
+    ProductCode = "f"
+    PackageCode = "10"
+
+
+class Win10InvAppFileValName(enum.Enum):
+    """
+    An enumeration containing the most useful Windows 10 Amcache
+    'InventoryApplicationFile' subkey value names.
+    """
+
+    FileId = "FileId"
+    LinkDate = "LinkDate"
+    LowerCaseLongPath = "LowerCaseLongPath"
+    ProductName = "ProductName"
+    ProductVersion = "ProductVersion"
+    ProgramID = "ProgramId"
+    Publisher = "Publisher"
+
+
+class Win10InvAppValName(enum.Enum):
+    """
+    An enumeration containing the most useful Windows 10 Amcache
+    'InventoryApplication' subkey value names.
+    """
+
+    InstallDate = "InstallDate"
+    Name = "Name"
+    Publisher = "Publisher"
+    RootDirPath = "RootDirPath"
+    Version = "Version"
+
+
+class Win10DriverBinaryValName(enum.Enum):
+    """
+    An enumeration containing the most useful Windows 10 Amcache
+    'InventoryDriverBinary' subkey value names.
+    """
+
+    DriverId = "DriverId"
+    DriverName = "DriverName"
+    DriverCompany = "DriverCompany"
+    Product = "Product"
+    Service = "Service"
+    DriverTimeStamp = "DriverTimeStamp"
+
+
+class AmcacheEntryType(enum.IntEnum):
+    Driver = 1
+    Program = 2
+    File = 3
+
+
+NullableString = Union[str, None, interfaces.renderers.BaseAbsentValue]
+NullableDatetime = Union[datetime.datetime, None, interfaces.renderers.BaseAbsentValue]
+
+
+@dataclasses.dataclass
+class AmcacheEntry:
+    """
+    A class containing all information about an entry from the Amcache registry hive.
+    Because all values could potentially be paged out of memory or malformed, they are all
+    a union between their expected value and `interfaces.renderers.BaseAbsentValue`.
+    """
+
+    entry_type: str
+    path: NullableString = renderers.NotApplicableValue()
+    company: NullableString = renderers.NotApplicableValue()
+    last_modify_time: NullableDatetime = renderers.NotApplicableValue()
+    last_modify_time_2: NullableDatetime = renderers.NotApplicableValue()
+    install_time: NullableDatetime = renderers.NotApplicableValue()
+    compile_time: NullableDatetime = renderers.NotApplicableValue()
+    sha1_hash: NullableString = renderers.NotApplicableValue()
+    service: NullableString = renderers.NotApplicableValue()
+    product_name: NullableString = renderers.NotApplicableValue()
+
+    def __post_init__(self):
+        # Loop through the fields
+        for field in dataclasses.fields(self):
+            if getattr(self, field.name) is None:
+                # Map 'None' values that have been explicitly set to `NotAvailableValue()`
+                setattr(self, field.name, renderers.NotAvailableValue())
```

Comment:
I worry that this is using renderer values (which are typically only for the renderer to figure out how to display them) in code that may be used by plugins (thus requiring the plugins to have to deal with them too).  Generally these values are only supplanted in over None values just before they're yielded into TreeGrid.  I'm happy that this is a separate class, and its only purpose may be for displaying output, but if it's like to get used in normal python code, then it may be better to stick to python's `None` type, and then making the yield statement, something like `this or renderers.NotAvailableValue`?